benchmark/main.js

```diff
@@ -0,0 +1,208 @@
+          logger.log(`${key.padStart(12)}: ${configuration[key]}`);
+        }
+      });
+      logger.groupEnd();
+      logger.group('Run');
+      let benchmark = new BenchmarkClass[configuration.modelClass](configuration.modelName, configuration.backend, configuration.iterations);
+      benchmark.onExecuteSingle = (i => logger.log(`Iteration: ${i + 1} / ${configuration.iterations}`));
+      let summary = await benchmark.runAsync();
+      benchmark.finalize();
+      logger.groupEnd();
+      if (summary.profilingResults && summary.profilingResults.length) {
+        logger.group('Profiling');
+        summary.profilingResults.forEach((line) => logger.log(line));
+        logger.groupEnd();
+      }
+      logger.group('Result');
+      logger.log(`Inference Time: <em style="color:green;font-weight:bolder;">${summary.computeResults.mean.toFixed(2)}+-${summary.computeResults.std.toFixed(2)}</em> [ms]`);
+      if (summary.decodeResults) {
+        logger.log(`Decode Time: <em style="color:green;font-weight:bolder;">${summary.decodeResults.mean.toFixed(2)}+-${summary.decodeResults.std.toFixed(2)}</em> [ms]`);
+      }
+      logger.groupEnd();
+    } catch (err) {
+      logger.error(err);
+    }
+    inputElement.removeAttribute('disabled');
+    pickBtnElement.setAttribute('class', 'btn btn-primary');
+    logger.groupEnd();
+  }
+
+  document.addEventListener('DOMContentLoaded', () => {
+    inputElement = document.getElementById('input');
+    pickBtnElement = document.getElementById('pickButton');
+    imageElement = document.getElementById('image');
+    canvasElement = document.getElementById('canvas');
+    showCanvasElement = document.getElementById('showCanvas');
+    segCanvas = document.getElementById('segCanvas');
+    inputElement.addEventListener('change', (e) => {
+      $('.labels-wrapper').empty();
+      let files = e.target.files;
+      if (files.length > 0) {
+        imageElement.src = URL.createObjectURL(files[0]);
+        bkImageSrc = imageElement.src;
+      }
+    }, false);
+    let modelElement = document.getElementById('modelName');
+    modelElement.addEventListener('change', (e) => {
+      $('.labels-wrapper').empty();
+      bkImageSrc = null;
+      let modelClass = modelElement.options[modelElement.selectedIndex].className;
+      let inputFile = document.getElementById('input').files[0];
+      if (inputFile !== undefined) {
+        imageElement.src = URL.createObjectURL(inputFile);
+      } else {
+        if (modelClass === 'skeleton_detection') {
+          imageElement.src = document.getElementById('poseImage').src;
+        } else if (modelClass === 'object_detection') {
+          imageElement.src = document.getElementById('objectDetectionImage').src;
+        } else if (modelClass === 'semantic_segmentation') {
+          imageElement.src = document.getElementById('segmentationImage').src;
+        } else {
+          imageElement.src = document.getElementById('imageClassificationImage').src;
```

Comment:
More use cases will be there, do you think the switch will be better than if-else- statement?